src/Compilers/CSharp/Test/Syntax/Parsing/LocalFunctionParsingTests.cs

```diff
@@ -2110,6 +2110,232 @@ void checkNodes()
             }
         }
 
+        [Fact, WorkItem(32106, "https://github.com/dotnet/roslyn/issues/32106")]
+        public void DuplicateAsyncs()
+        {
+            const string text = """
+                class Program
+                {
+                    void M()
+                    {
+                        async async void F1() { }
```

Comment:
>async async void F1() { } [](http://example.com/codeflow?start=24&length=25)

Please place each scenario into its own test, so that reviewers could actually make sense of what is going on.